### ISSUE TITLE
DDP

### ISSUE THREAD

**Issue by onlinehuazai** [pytorch/pytorch#62403]
https://pytorch.org/docs/stable/notes/ddp.html

import torch
import torch.distributed as dist
import torch.multiprocessing as mp
import torch.nn as nn
import torch.optim as optim
from torch.nn.parallel import DistributedDataParallel as DDP

def example(rank, world_size):
    # create default process group
    dist.init_process_group("gloo", rank=rank, world_size=world_size)
    # create local model
    model = nn.Linear(10, 10).to(rank)
    # construct DDP model
    ddp_model = DDP(model, device_ids=[rank])
    # define loss function and optimizer
    loss_fn = nn.MSELoss()
    optimizer = optim.SGD(ddp_model.parameters(), lr=0.001)

    # forward pass
    outputs = ddp_model(torch.randn(20, 10).to(rank))
    labels = torch.randn(20, 10).to(rank)
    # backward pass
    loss_fn(outputs, labels).backward()
    # update parameters
    optimizer.step()

def main():
    world_size = 2
    mp.spawn(example,
        args=(world_size,),
        nprocs=world_size,
        join=True)

if __name__=="__main__":
    main()

Exception: process 1 terminated with exit code 1

**Comment by heitorschueroff** [comment]
Could you please fix the formatting of your code and add some more detail about what you are trying to do and what you'd expect the correct behavior to be? That would help us help you more effectively.

**Comment by onlinehuazai** [comment]
> Could you please fix the formatting of your code and add some more detail about what you are trying to do and what you'd expect the correct behavior to be? That would help us help you more effectively.

code is from https://pytorch.org/docs/stable/notes/ddp.html, error is Exception: process 1 terminated with exit code 1
